### PR TITLE
Issue #54 save load macro from file

### PR DESCRIPTION
**Pull request by NCKiser**
Added two buttons in bottom-left of the macro tab to save the current macro to a file, and import a macro from a file.

### CODE REVIEW

**Comment by NCKiser** [comment]
Issue #54

File: src/main/python/macro_tab.py [hunk 1]
```diff
@@ -122,6 +130,52 @@ def on_move(self, obj, offset):
...
+                "text": ActionText,
+                "delay": ActionDelay,
+            }
+            tag_to_actionUI = {
+                "down": ActionDownUI,
+                "up": ActionUpUI,
+                "tap": ActionTapUI,
+                "text": ActionTextUI,
+                "delay": ActionDelayUI,
+            }
```
Comment (xyzz) [inline src/main/python/macro_tab.py:166]: Can you reduce the duplication here, there's already a similar implementation in `restore_macros` which restores from json into a macro (that specific part could be moved into a separate utility function); and then a mapping of action to UI element is available in macro_recorder.py->ui_action.

Still not ideal and macro code overall is messy, but what I want to ensure is that when adding another macro type that saving/restoring functionality doesn't have to be duplicated in yet another place.

**Review by xyzz — CHANGES_REQUESTED** [review]

File: src/main/python/macro_tab.py (hunk 1 above)
Comment (NCKiser) [reply src/main/python/macro_tab.py:166]: That makes sense, I will work on reducing the duplication. KISS

Comment (NCKiser) [reply src/main/python/macro_tab.py:166]: https://en.wikipedia.org/wiki/KISS_principle

Comment (NCKiser) [reply src/main/python/macro_tab.py]: Commit baf603a addresses this change

**Pull request merged**